src/uid.js

```diff
@@ -0,0 +1,5 @@
+function autoInc(seed = 0) {
```

Comment:
does babel understand generators? If so we could rewrite this as:
```
function* autoInc() {
  var uid = 0;
  while (true)
    yield uid++;
}
```
rather than exporting a function that returns a function.

Reply:
Babel understands generator and can transpile them just fine but transpiled version is not nice - its a machine state which has switch statement in its body for each state which can be understood by regenerator-runtime and its way too big for this simple case unfortunately.

U can check transpiled generators in babels online repl and check out regenerator-runtime's code in their repo

Reply:
https://babeljs.io/repl/#?babili=false&browsers=ie%2011&build=&builtIns=false&circleciRepo=&code_lz=GYVwdgxgLglg9mAVAAgIYinAkpAFASmQG8AoZZAN1QCdkQYATZAXmQAYBuM5AdwAsYAGwCmyXFGohh-buQCeMYYKb0GAajVcAvkA&debug=false&evaluate=false&lineWrap=false&presets=es2015&prettier=false&showSidebar=true&targets=&version=6.26.0

+ this needs to be included for it to work - https://github.com/facebook/regenerator/blob/30d34536b9e3f7a2873b04a16ec66fec9c8246f6/packages/regenerator-runtime/runtime.js